src/renderers/shared/fiber/ReactFiberTreeReflection.js

```diff
@@ -114,3 +114,14 @@ exports.findCurrentHostFiber = function(component : ReactComponent<any, any, any
   }
   return null;
 };
+
+exports.getComponentName = function(fiber: Fiber): string {
+  const type = fiber.type;
+  const instance = fiber.stateNode;
+  const constructor = instance && instance.constructor;
+  return (
+    type.displayName || (constructor && constructor.displayName) ||
+    type.name || (constructor && constructor.name) ||
+    'A Component'
+  );
+};
```

Comment:
I tried but it was intentionally DEV-only at the time, and I need it for an invariant.
If we're ok increasing non-Fiber size with Fiber code before Fiber is ready, I could remove the DEV gate there.